apis/github

```diff
@@ -76,51 +76,84 @@ Blob.fullPath = function(self)
 		return self.path
 	end
 end
+Blob.getContents = function(self)
+	local data = http.get(
+		('https://raw.github.com/%s/%s/%s/%s'):format(
+			self.repo.user, self.repo.name, self.sha,
+			encodeURI(self:fullPath())
+		)
+	)
+	return data.readAll()
+end
 
 -- A class for a tree (aka a folder)
 local Tree = {}
 Tree.__index = Tree
 Tree.new = function(repo, sha, path)
-	local url = ('repos/%s/%s/git/trees/%s'):format(repo.user, repo.name, sha)
-	local status, data = getAPI(url, repo.auth)
+	return setmetatable({
+		repo = repo,
+		sha = sha,
+		path = path or ''
+	}, Tree)
+end
+Tree.getContents = function(self)
+	if self.contents then
+		return self.contents
```

Comment:
Can we make this look more private?

At risk of revealing that I now develop python and not lua, perhaps

```suggestion
	if self._contents then
		return self._contents
```

Reply:
Also, will this fail for empty files? Should it be `contents ~= nil`?

Reply:
Sure, private change committed in 86fb962.

That test will work fine for empty directories - an empty table is a truthy value, unlike `nil`.